packages/base-crypto/src/VirgilCrypto.ts

```diff
@@ -256,11 +402,22 @@ export class VirgilCrypto implements ICrypto {
     validatePublicKey(publicKey);
 
     const verifier = new this.foundationModules.Verifier();
-
-    verifier.reset(mySignature);
+    try {
+      verifier.reset(mySignature);
+    } catch (error) {
+      verifier.delete();
+      throw error;
+    }
     verifier.appendData(myData);
 
-    const lowLevelPublicKey = importPublicKey(publicKey.key);
+    let lowLevelPublicKey: LowLevelPublicKey | undefined;
+    try {
+      lowLevelPublicKey = importPublicKey(publicKey.key);
```

Comment:
If you put this before the creating the `Verifier`, you won't need a `try..catch` block around it